### PR TITLE
Azure pipelines

### LINKED ISSUE #9: Have pre-built binary releases like fd, ripgrep, etc

### PR DESCRIPTION
**Pull request by chmln**
- [x] Run tests 
- [x] Release binaries
- [x] build.rs
  - [x] Generate completions
  - [x] Generate man pages

Resolves #9, #23, #30

### CODE REVIEW

**Comment by xoac** [comment]
Hi I saw your commit on reddit. Here is preRelase for v0.0.2 that support github releases! 

I would love feedback: It looks like this: https://github.com/xoac/rust-azure-pipelines/releases

You just need one file `azure-piplines.yml` https://github.com/xoac/rust-azure-pipelines/blob/v0.0.2-rc/azure-pipelines.yml

Please change:
```
 ref: refs/heads/master ## TODO you may want to change it to refs/tags/TAG_NAME. 
```
to
```
 ref: refs/tags/v0.0.2-rc
```

**Comment by xoac** [comment]
Here is build for tag release https://dev.azure.com/sylwesterrapala/azure-piplines/_build/results?buildId=131

**Comment by chmln** [comment]
Hey @xoac and thanks a lot for your efforts.

I took a look at the `azure-pipelines.yml` file and looks like it doesn't build MacOS binaries, which I absolutely need. Did I miss something or its just not there yet?

**Comment by xoac** [comment]
If cross dosent support it that is true. But I can use macOS agent and add check for mac target and build it without cross.

**Comment by chmln** [comment]
@xoac that would be amazing.

**Comment by xoac** [comment]
@chmln It's now in master. Please check it satisfy you.

https://github.com/xoac/rust-azure-pipelines/commit/266ea3dad393acaa27d49694d3542e72d0a97820

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    include: ['*']
+  tags:
+    include: ['*']
+
+resources:
+  repositories:
+    - repository: templates
+      type: github
+      name: xoac/rust-azure-pipelines
+      ref: refs/heads/master ## TODO you may want to change it to refs/tags/TAG_NAME.
```
Comment (xoac) [inline azure-pipelines.yml]: You probably wanna change it from master to `/tags/v0.0.2` since I can break master any time

Comment (xoac) [reply azure-pipelines.yml]: @chmln (just to make sure you will read this)

**Pull request merged**